valqueries-automapper/src/main/java/com/valqueries/automapper/ValqueriesQueryImpl.java

```diff
@@ -158,16 +158,23 @@ public String buildSelectSql(String tableAlias, String... columns) {
 			}
 		}
 		String sql ="SELECT " + columnsSql + eagerSelect + " FROM (SELECT * FROM " + getTableName(Clazz.of(typeDescriber.clazz())) + " " + tableAlias + " " + elements.stream().map(Element::fromString).filter(Objects::nonNull).collect(Collectors.joining(", "));
+
 		if (!elements.isEmpty()) {
 			sql += " WHERE " + elements.stream().map(Element::queryString).collect(Collectors.joining(" AND "));
 		}
+
 		if (!sortElements.isEmpty()) {
 			sql += " ORDER BY " + sortElements.stream().map(Element::queryString).collect(Collectors.joining(", "));
 		}
+
 		if (limit !=  null) {
 			sql += dialect.getLimitDefinition(offset, limit);
 		}
 
+		if(!sortElements.isEmpty() && limit ==null){
```

Comment:
Can you do it in else-if manner, so that there wouldn't be an impression that we can add limitdefinition twice